ros2topic/ros2topic/verb/delay.py

```diff
@@ -186,218 +152,90 @@ def get_delay(self):
-    if index_string_parts[0] != '':
-        try:
-            slice_args[0] = int(index_string_parts[0])
-        except ValueError:
-            assert False, "non-integer slice start '%s'" % index_string_parts[0]
-    if index_string_parts[1] != '':
-        try:
-            slice_args[1] = int(index_string_parts[1])
-        except ValueError:
-            assert False, "non-integer slice stop '%s'" % index_string_parts[1]
-    if len(index_string_parts) > 2 and index_string_parts[2] != '':
-            try:
-                slice_args[2] = int(index_string_parts[2])
-            except ValueError:
-                assert False, "non-integer slice step '%s'" % index_string_parts[2]
-    if len(index_string_parts) > 3:
-        assert False, 'too many slice arguments'
-    return slice(*slice_args)
-
-def _get_nested_attribute(msg, nested_attributes):
-    value = msg
-    for attr in nested_attributes.split('/'):
-        value = getattr(value, attr)
-    return value
-
-def _get_topic_type(topic):
-    """
-    subroutine for getting the topic type
-    :returns: topic type, real topic name and fn to evaluate the message instance
-    if the topic points to a field within a topic, e.g. /rosout/msg, ``(str, str, fn)``
-    """
-    try:
-        val = _master_get_topic_types(rosgraph.Master('/rostopic'))
-    except socket.error:
-        raise ROSTopicIOException("Unable to communicate with master!")
-
-    # exact match first, followed by prefix match
-    matches = [(t, t_type) for t, t_type in val if t == topic]
-    if not matches:
-        matches = [(t, t_type) for t, t_type in val if topic.startswith(t+'/')]
-        # choose longest match
-        matches.sort(key=itemgetter(0), reverse=True)
-
-        # try to ignore messages which don't have the field specified as part of the topic name
-        while matches:
-            t, t_type = matches[0]
-            msg_class = roslib.message.get_message_class(t_type)
-            if not msg_class:
-                # if any class is not fetchable skip ignoring any message types
-                break
-            msg = msg_class()
-            nested_attributes = topic[len(t) + 1:].rstrip('/')
-            nested_attributes = nested_attributes.split('[')[0]
-            if nested_attributes == '':
-                break
-            try:
-                _get_nested_attribute(msg, nested_attributes)
-            except AttributeError:
-                # ignore this type since it does not have the requested field
-                matches.pop(0)
-                continue
-            matches = [(t, t_type)]
-            break
+    def timer_callback():
+        rt.print_delay()
 
-    if matches:
-        t, t_type = matches[0]
-        if t_type == rosgraph.names.ANYTYPE:
-            return None, None, None
-        return t_type, t, msgevalgen(topic[len(t):])
-    else:
-        return None, None, None
+    timer = node.create_timer(1, timer_callback)
```

Comment:
You could register `rt.print_delay` here directly getting rid of the local function.

Reply:
fixed 9c6ad41